deepchem/models/keras_model.py

```diff
@@ -426,20 +474,50 @@ def fit_generator(self,
         avg_loss = 0.0
         averaged_batches = 0
 
+      # Calculate epoch number, sample count, and metrics
+      epoch_num = self._get_epoch_num(current_step)
+      sample_count = self._get_sample_count(current_step)
+
+      # Decide whether to calculate metrics at this current step
+      scores = None
+      if self.metrics is not None and self.metrics:
+        if (self.logging_strategy == "step" and current_step % self.log_frequency == 0) or \
+           (self.logging_strategy == "epoch" and current_step % steps_per_epoch == 0):
+          scores = self.evaluate(self.dataset, self.metrics)
```

Comment:
I don't think this is the right way or place to be computing metrics.  If you're computing metrics, you don't necessarily want to compute them on the training set (which again might not even be a Dataset object), and you might not want to compute them at the same interval at which you record training loss (since this is much more expensive).

Reply:
In your opinion, what do you think is the best approach to log metrics for training? I know that some users prefer to see metrics being calculated on the training dataset during training too. From what I understand, there are no metrics that are passed into `fit()` originally so there would need to be a place to pass in what metrics to log.